test/generic/pointers/test_model_pointer.py

```diff
@@ -0,0 +1,23 @@
+import torch
+from torch import nn
+import syft as sy
+
+
+def test_copy_on_remote():
+    """
+        Tests the correctness of copying a remote model.
+    """
+    data = torch.tensor([[0, 0], [0, 1], [1, 0], [1, 1.0]])
+    model = nn.Linear(2, 1)
+
+    pred_local = model(data)
+
+    hook = sy.TorchHook(torch)
+    bob = sy.VirtualWorker(hook, id="bob")
+    data_bob = data.send(bob)
+    model_bob = model.send(bob)
+    model_bob_copy = model_bob.copy()
+    pred_remote = model_bob_copy(data_bob)
+    pred_remote = pred_remote.get()
```

Comment:
```suggestion
    model_bob.get()
    pred_remote = model_bob_copy(data_bob)
    pred_remote = pred_remote.get()
```

Reply:
The suggested change introduces an error. I investigated and this is what I think is happening: 
1) There is no straightforward way to copy a model in torch. 
2) Upon trying to copy a model in soft under the hood you actually call clone. 
3) The difference is that (from the code): Clone should keep ids unchanged, contrary to copy.
4) Therefore we are actually creating two pointers to the same location. Once we "get" the object, it releases the other pointer as well.
5) Tensor on the other hand, do allow copy(). This code will work as expected:   
```
t = torch.tensor([3, 1.0, 2])
p1 = t.send(alice)
p2 = p1.copy()
p1.get()
p2.get()
```
6) In the example from 5 If we replace the `copy()` with `clone()` the same error will occur since p2 will have the same id as p1.